homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -65,18 +70,32 @@ async def async_step_user(self, user_input=None):
                 errors["base"] = "cannot_connect"
             except InvalidAuth:
                 errors["base"] = "invalid_auth"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
             else:
-                await self.async_set_unique_id(info["serial"])
+                await self.async_set_unique_id(info[KEY_SYS_SERIAL])
                 self._abort_if_unique_id_configured()
-                return self.async_create_entry(title=info["title"], data=user_input)
+                return self.async_create_entry(
```

Comment:
If we're reauthenticating, we should not create a new config entry, but abort with success reason.

https://github.com/home-assistant/core/blob/343e8f0ecc215798a3e3892ba63b964ce5b72cc1/homeassistant/strings.json#L129

Reply:
I cribbed the `async_step_reauth_confirm` flow from some other integrations, so that it doesn't call into `async_step_user` anymore. Please let me know if this is OK now.

And I added the missing strings.json reauth entries.

(in https://github.com/home-assistant/core/pull/99411)